core/src/main/java/com/linecorp/armeria/server/RoutingTrie.java

```diff
@@ -225,6 +288,33 @@ private void findAllNodes(Node<V> node, String path, int begin, boolean exact,
         return continueWalking();
     }
 
+    @Nullable
+    private Node<V> filterAcceptableNode(Node<V> node,
+                                         RoutingContext routingCtx,
+                                         boolean isRouteDecorator,
+                                         @Nullable ImmutableList.Builder<Routed<V>> routeAccumulator) {
+        final List<Routed<V>> routingResults =
+                node.values.stream()
+                           // TODO(hyangtack) Add dryRun parameter to Route.apply
+                           //                 Pass it as true if routeAccumulator is null
+                           //                 not to set deferred exception to context.
+                           .map(v -> {
+                               final Route route = routeResolver.apply(v);
+                               final RoutingResult result = route.apply(routingCtx, isRouteDecorator);
+                               return result.isPresent() ? Routed.of(route, result, v) : null;
```

Comment:
Do I need to add `dryRun` mode to `Route` to avoid setting deferred exception to `RoutingContext`?
I guess it's not necessary because we use `RoutingTrie` only for routing a request. Please let me know if I missed something. :-)